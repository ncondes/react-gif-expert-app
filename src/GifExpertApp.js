import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = () => {

  const defaultCategories = [ 'Los Simpsons' ];
  const [ categories, setCategories ] = useState( defaultCategories );

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory
        setCategories={ setCategories }
      />
      <hr />


      { categories.map( category => (
        <GifGrid
          category={ category }
          key={ category }
        />
      ))}


    </>
  )
}

