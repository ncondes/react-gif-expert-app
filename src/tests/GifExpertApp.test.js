import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en GifExpertApp', () => {

  test('Deberia hacer match con el snapshot', () => {
    
    const wrapper = shallow( <GifExpertApp /> );
    
    expect( wrapper ).toMatchSnapshot();

  });

  test('Deberia de mostrar una lista de categorias', () => {
    
    const categories = [ 'One punch', 'Dragon ball' ];
    const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />);

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );

  })
  
})