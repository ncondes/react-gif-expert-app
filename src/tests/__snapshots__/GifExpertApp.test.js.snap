// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Deberia de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One punch"
    key="One punch"
  />
  <GifGrid
    category="Dragon ball"
    key="Dragon ball"
  />
  <p
    className="author"
  >
    Coded by Nicolas Conde
  </p>
</Fragment>
`;

exports[`Pruebas en GifExpertApp Deberia hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <p
    className="author"
  >
    Coded by Nicolas Conde
  </p>
</Fragment>
`;
